### PR TITLE
[7.48.x]JBPM-9273 System error occurred while adding deployment unit

### PR DESCRIPTION
**Pull request by bxf12315**
**Thank you for submitting this pull request**

**JIRA**: [JBPM-9273](https://issues.redhat.com/browse/JBPM-9273)

**Referenced Pull Requests**:
	https://github.com/kiegroup/droolsjbpm-integration/pull/2381

**Business Central**: [WAR file](https://donwnload.here/something)

**VS Code**: [plugin](https://donwnload.here/something)

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* Retest PR: <b>jenkins retest this</b>
* A full downstream build: <b>jenkins do fdb</b>
* A compile downstream build: <b>jenkins do cdb</b>
* A full production downstream build: <b>jenkins do product fdb</b>
* An upstream build: <b>jenkins do upstream</b>
</details>

### CODE REVIEW

**Comment by bxf12315** [comment]
dependent on https://github.com/kiegroup/droolsjbpm-integration/pull/2381

**Comment by prashuts** [comment]
jenkins do fdb

**Review by prashuts — APPROVED** [review]
LGTM

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/container/ContainerPresenterTest.java [hunk 1]
```diff
@@ -196,6 +196,7 @@ public void testInit() {
     @Test
     public void testStartContainer() {
         presenter.loadContainers(containerSpecData);
+
```
Comment (nmirasch) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/container/ContainerPresenterTest.java:199]: @bxf12315 is this change needed? :-)

**Review by nmirasch — APPROVED** [review]

**Review by sutaakar — APPROVED** [review]

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/container/ContainerPresenterTest.java (hunk 1 above)
Comment (bxf12315) [reply kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/test/java/org/kie/workbench/common/screens/server/management/client/container/ContainerPresenterTest.java]: Done

**Comment by bxf12315** [comment]
jenkins do fdb

**Comment by prashuts** [comment]
LGTM

**Comment by bxf12315** [comment]
jenkins do fdb

**Comment by bxf12315** [comment]
jenkins retest this

**Comment by prashuts** [comment]
jenkins do fdb

**Comment by bxf12315** [comment]
jenkins do fdb

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/container/ContainerPresenter.java [hunk 2]
```diff
@@ -147,13 +151,34 @@ public void loadContainers(@Observes final ContainerSpecData content) {
...
+        });
+    }
+
+    private boolean isFailedContainerSpec(Collection<Container> containers, ContainerSpec containerSpec) {
+        Optional<Container> allContainersAreFailed = containers.stream().filter(container -> KieContainerStatus.FAILED != container.getStatus()).findFirst();
+        if (!allContainersAreFailed.isPresent() && containers.size() > 0) {
+            return true;
+        }
+        return false;
+    }
```
Comment (Ginxo) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/container/ContainerPresenter.java:180]: I would refactor this method by
```
private boolean isFailedContainerSpec(Collection<Container> containers, ContainerSpec containerSpec) {
        return !CollectionUtils.isEmpty(containers) && !containers.stream().filter(container -> KieContainerStatus.FAILED != container.getStatus()).findFirst().isPresent()
    }
```
this way you avoid NullPointerException from `containers.stream` and it's simpler to read (FMPOV)

File: kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/container/ContainerPresenter.java [hunk 3]
```diff
@@ -147,13 +151,34 @@ public void loadContainers(@Observes final ContainerSpecData content) {
...
+                content.getContainerSpec().setStatus(KieContainerStatus.FAILED);
+            }
             setup(content.getContainerSpec(),
                   content.getContainers());
         } else {
             logger.warn("Illegal event argument.");
         }
     }
 
+    private void resetReleaseIdForFailedContainers(Collection<Container> containers, ContainerSpec containerSpec) {
```
Comment (Ginxo) [inline kie-wb-common-screens/kie-wb-common-server-ui/kie-wb-common-server-ui-client/src/main/java/org/kie/workbench/common/screens/server/management/client/container/ContainerPresenter.java:165]: What if you do this instead

```
    private void resetReleaseIdForFailedContainers(Collection<Container> containers, ContainerSpec containerSpec) {
        if(!CollectionsUtils.isEmpty(containers)) {
           containers.stream().
           .filter(container -> KieContainerStatus.FAILED == container.getStatus() || container.getResolvedReleasedId() == null)
           .forEach(container -> {
                container.setResolvedReleasedId(containerSpec.getReleasedId());
                container.addMessage(new Message(Severity.ERROR, Collections.emptyList()));
            });
        }
    }
```
it doesn't change much, but keeps functional programming approach in the whole method, but up to you, just proposing...

**Review by Ginxo — APPROVED** [review]
LGTM, just few technical proposals
I know nothing about functionality here...

**Comment by prashuts** [comment]
@bxf12315  @Ginxo - M2_HOME env variable added in "github/workflows/pull_request.yml " file and the path is hardcoded. is it okay to be like this?
@bxf12315 i hope this is final build ..(lol)
Other than this looks good to me..

**Comment by bxf12315** [comment]
jenkins do fdb

**Comment by bxf12315** [comment]
> @bxf12315 @Ginxo - M2_HOME env variable added in "github/workflows/pull_request.yml " file and the path is hardcoded. is it okay to be like this?
> @bxf12315 i hope this is final build ..(lol)
> Other than this looks good to me..

It just want to fix our jdk8 build failed issue, I also hope it's the last one.

**Review by prashuts — APPROVED** [review]

**Comment by mareknovotny** [comment]
Even I agree with @Ginxo's comments, this is backport of the changes from master, so we should refactor those 2 methods in master and probably leave 7.48.x branch PR as it is for now.
